sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client_models.hpp

```diff
@@ -29,34 +30,18 @@ namespace Azure { namespace Security { namespace Attestation { namespace Models
    * the attestation service.
    *
    */
-  class AttestationType final {
-  private:
-    std::string m_attestationType;
-
+  class AttestationType final
+      : public Azure::Core::_internal::ExtendableEnumeration<AttestationType> {
   public:
     /**
      * @brief Construct a new AttestationType object
      *
      * @param attestationType The string attestationType used for the attestation policy operation.
      */
-    AttestationType(std::string attestationType) : m_attestationType(std::move(attestationType)) {}
-
-    /**
-     * @brief Enable comparing the ext enum.
-     *
-     * @param other Another #AttestationType to be compared.
-     */
-    bool operator==(AttestationType const& other) const
+    AttestationType(std::string attestationType) : ExtendableEnumeration(std::move(attestationType))
```

Comment:
This constructor should probably be `explicit` for type safety; we don't want `f(AttestationType)` called as `f(some_string)`.

Reply:
Done.